src/App/Entity/AbstractDiscussion.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+namespace App\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Validator\Constraints as Assert;
+use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
+use Gedmo\Mapping\Annotation as Gedmo;
+use Doctrine\Common\Collections\ArrayCollection;
+
+/**
+ * @ORM\Entity
+ * @ORM\InheritanceType("SINGLE_TABLE")
+ * @ORM\DiscriminatorColumn(name="discr", type="string")
+ * @ORM\DiscriminatorMap({"abstract_discussion" = "AbstractDiscussion", "public_discussion" = "PublicDiscussion", "private_discussion" = "PrivateDiscussion"})
+ * @ORM\HasLifecycleCallbacks
+ * @UniqueEntity(fields="slug")
+ */
+class AbstractDiscussion
+{
+    /**
+     * @ORM\Column(type="integer", unique=true)
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @Gedmo\Slug(fields={"title"})
+     * @ORM\Column(length=128, unique=true)
+     */
+    protected $slug;
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     * @Assert\Length(
+     *      min = "2",
+     *      max = "255"
+     * )
+     */
+    protected $title;
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     * @Assert\NotBlank()
+     */
+    protected $abstract;
+
+    /**
+     * @ORM\Column(type="datetime")
+     * @Assert\DateTime()
+     */
+    protected $creationDate;
+
+    /**
+     * @ORM\Column(type="datetime")
+     * @Assert\DateTime()
+     */
+    protected $lastEditDate;
+
+    /**
+     * If the discussion is locked the users cannot comment but can still access it.
+     *
+     * @ORM\Column(type="boolean")
+     */
+    protected $locked;
+
+    /**
+     * @ORM\OneToMany(targetEntity="Post", mappedBy="discussion", cascade={"persist", "remove"})
+     * @Assert\Valid()
+     */
+    protected $posts;
+
+    /**
+     * @ORM\ManyToMany(targetEntity="User", inversedBy="unreadDiscussions", cascade={"persist"})
+     * @ORM\JoinTable(name="discussions_unreaders")
+     * @Assert\Valid()
+     */
+    protected $unreaders;
+
+    public function __construct()
+    {
+        $this->locked = false;
+        $this->posts  = new ArrayCollection();
+        $this->unreaders = new ArrayCollection();
+    }
+
+    public function setSlug($slug)
+    {
+        $this->slug = $slug;
+
+        return $this;
+    }
+
+    public function getSlug()
+    {
+        return $this->slug;
+    }
+
+    public function setTitle($title)
+    {
+        $this->title = $title;
+
+        return $this;
+    }
+
+    public function getTitle()
+    {
+        return $this->title;
+    }
+
+    public function setAbstract($abstract)
+    {
+        $this->abstract = $abstract;
+
+        return $this;
+    }
+
+    public function getAbstract()
+    {
+        return $this->abstract;
+    }
+
+    /**
+     * @ORM\PrePersist()
+     */
+    public function setCreationDate()
```

Comment:
Why would you have a setter for this? Set this property directly in the constructor instead.